wasm/store.go

```diff
@@ -844,6 +847,7 @@ func DecodeBlockType(types []*TypeInstance, r io.Reader) (*FunctionType, uint64,
 
 // HostFunctionCallContext is the first argument of all host functions.
 type HostFunctionCallContext struct {
+	context.Context
```

Comment:
I think we don't typically use struct embedding, so we should trailing comment why if we do. Alternatively, we can de-embed this. wdyt @pkedy @mathetake @nullpo-head?
```suggestion
	context.Context
```

Reply:
yeah without concrete reason, we would like to avoid embedding

Reply:
but this case seems reasonable to me so trailing comments would help

Reply:
Embedding context I think is pretty common in web frameworks so you can pass it along. The standard http library does not so it might be a matter of preference. What are the reasons not to embed it? Just curious 😀

Reply:
Clarifying the owner of fields/methods at callsite is one!

Reply:
Would you prefer a `Context()` func like in the http standard library?

Reply:
Please go for that one `Context()` as this type will be redone anyway I think and it won't be a field. The pros and cons of struct embedding at that point become internal details https://github.com/tetratelabs/wazero/issues/204

sg @mathetake ?

Reply:
sg!

Reply:
Nice. I have the change ready. Committing now.

Reply:
I will say that it will be a little weird that `Context()` is a func and `Memory` is a field. But as you said, this will be reworked anyway.

Reply:
Ack and I promise I'll do my best to unweird it soon

Reply:
also PS I'm not doing this now because I don't want to drift your PR :D